.circleci/config.yml

```diff
@@ -386,13 +389,25 @@ commands:
           name: Install <<parameters.extras-requires>>
           command: |
             if [ -n "<<parameters.extras-requires>>" ]; then
-              pip install --progress-bar off <<parameters.extras-requires>>
+              i=0
+              while true; do
```

Comment:
a script based retry is probably the best solution here, but would it make sense to break this out to it's own file / command so that it can easily be reused? could take number of retries as an environment based input and command to retry as the command arguments

Reply:
I was thinking about having a separate script but wasn't sure how that would be organized or installed here. Any idea? A command might not be flexible enough here since it needs to be called in a bash script.

Reply:
You can add the new script in `.circleci/scripts`.

Reply:
if we had a retry script like https://gist.github.com/erikwilson/cf63688b12f89a68a6397d9605b8d18e in `tools/scripts` I think that might be sufficient. Alternatively it could live in `.circleci/scripts`, but it would be nice to have it available for make also. The PATH could then be modified so retry is available, or the script location/call could be explicit, or alternatively a RETRY_CMD path could be passed around with ${RETRY_CMD} prepended to other commands (where an empty environment variable does not actually retry).

Reply:
Cool, will use one of the directories mentioned and update the PR.